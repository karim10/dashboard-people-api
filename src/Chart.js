import React from 'react';
import { BarChart, Bar, XAxis, YAxis, Label, ResponsiveContainer } from 'recharts';
import Title from './Title';
import PropTypes from 'prop-types';

export default function Chart({ chartType, chartData }) {
    let data = chartData;
    // displays randomly 7 countries scores
    if (chartType === 'country') {
        data = chartData.sort(() => Math.random() - Math.random()).slice(0, 7);
    }
    return (
        <React.Fragment>
            <Title>Score statistics</Title>
            <ResponsiveContainer>
                <BarChart
                    data={data}
                    margin={{
                        top: 16,
                        right: 16,
                        bottom: 0,
                        left: 24
                    }}
                >
                    <XAxis dataKey={chartType} />
                    <YAxis>
                        <Label angle={270} position="left" style={{ textAnchor: 'middle' }}>
                            Average score
                        </Label>
                    </YAxis>
                    <Bar type="monotone" dataKey="score" fill="#556CD6" />
                </BarChart>
            </ResponsiveContainer>
        </React.Fragment>
    );
}

Chart.propTypes = {
    chartType: PropTypes.oneOf(['country', 'gender']).isRequired,
    chartData: PropTypes.array.isRequired
};
