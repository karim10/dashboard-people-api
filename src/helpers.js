export const getChartDataByCountry = peopleData => {
    const groupByCountryObj = peopleData.reduce((acc, person) => {
        if (acc.hasOwnProperty(person.country)) {
            acc[person.country].totalScore = +person.score;
            acc[person.country].totalPeople++;
        } else {
            acc[person.country] = { totalScore: person.score, totalPeople: 1 };
        }
        return acc;
    }, {});

    const groupByCountryArray = Object.keys(groupByCountryObj).map(country => ({
        country,
        score: groupByCountryObj[country].totalScore / groupByCountryObj[country].totalPeople
    }));
    return groupByCountryArray;
};

const getAverageScoreByGender = (groupByGenderObj, gender) =>
    groupByGenderObj[gender].totalScore / groupByGenderObj[gender].totalPeople;

const initAccValue = {
    totalScore: 0,
    totalPeople: 0
};

export const getChartDataByGender = peopleData => {
    const groupByGenderObj = peopleData.reduce(
        (acc, person) => {
            if (person.gender === 'Female') {
                acc.female.totalScore += person.score;
                acc.female.totalPeople++;
            } else if (person.gender === 'Male') {
                acc.male.totalScore += person.score;
                acc.male.totalPeople++;
            } else {
                acc.NA.totalScore += person.score;
                acc.NA.totalPeople++;
            }
            return acc;
        },
        {
            female: { ...initAccValue },
            male: { ...initAccValue },
            NA: { ...initAccValue }
        }
    );
    const groupByGenderArray = [
        {
            gender: 'Female',
            score: getAverageScoreByGender(groupByGenderObj, 'female')
        },
        {
            gender: 'Male',
            score: getAverageScoreByGender(groupByGenderObj, 'male')
        },
        {
            gender: 'NA',
            score: getAverageScoreByGender(groupByGenderObj, 'NA')
        }
    ];
    return groupByGenderArray;
};

// compares two strings with omitting null values
function compareStrings(item1, item2, orderDirection, orderBy) {
    if (item1[orderBy] === item2[orderBy]) {
        return 0;
    } else if (item1[orderBy] === null) {
        return 1;
    } else if (item2[orderBy] === null) {
        return -1;
    } else if (orderDirection === 'asc') {
        return item1[orderBy].toLowerCase() < item2[orderBy].toLowerCase() ? 1 : -1;
    } else {
        return item1[orderBy].toLowerCase() < item2[orderBy].toLowerCase() ? -1 : 1;
    }
}

function compareNumericals(item1, item2, orderDirection, orderBy) {
    return orderDirection === 'asc'
        ? item1[orderBy] - item2[orderBy]
        : item2[orderBy] - item1[orderBy];
}

export function getSortedTable(peopleData, { orderDirection, orderBy, isNumeric }) {
    return peopleData.sort((item1, item2) =>
        isNumeric
            ? compareNumericals(item1, item2, orderDirection, orderBy)
            : compareStrings(item1, item2, orderDirection, orderBy)
    );
}
const fileteringAttributes = ['first_name', 'last_name', 'gender', 'city', 'country', 'score'];
export function getFilteredTable(peopleData, queryInput) {
    return peopleData.filter(person =>
        Object.keys(person).reduce((acc, att) => {
            if (person[att] === null || fileteringAttributes.indexOf(att) === -1) {
                return acc;
            }
            acc =
                person[att]
                    .toString()
                    .toLowerCase()
                    .indexOf(queryInput.toLowerCase()) !== -1
                    ? true
                    : acc;
            return acc;
        }, false)
    );
}
