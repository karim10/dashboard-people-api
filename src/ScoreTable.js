/* eslint-disable no-script-url */

import React from "react";
import Link from "@material-ui/core/Link";
import { makeStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import TableSortLabel from "@material-ui/core/TableSortLabel";
import Title from "./Title";

import { getSortedTable } from "./helpers";

const useStyles = makeStyles(theme => ({
  seeMore: {
    marginTop: theme.spacing(3)
  },
  tableColumn: {
    width: "18%"
  }
}));

const columnsDetails = [
  {
    label: "Last Name",
    attribute: "last_name",
    attributeType: "string"
  },
  {
    label: "First Name",
    attribute: "first_name",
    attributeType: "string"
  },
  {
    label: "Gender",
    attribute: "gender",
    attributeType: "string"
  },
  {
    label: "City",
    attribute: "city",
    attributeType: "string"
  },
  {
    label: "Country",
    attribute: "country",
    attributeType: "string"
  },
  {
    label: "Score",
    attribute: "score",
    attributeType: "number"
  }
];

function EnhancedHeaderTable(props) {
  const classes = useStyles();
  const { columnsDetails, handleSorting, order, orderBy } = props;
  return (
    <TableHead>
      <TableRow>
        {columnsDetails.map((columnDetails, i) => {
          const alignment = i === columnsDetails.length - 1 ? "right" : "left";
          return (
            <TableCell
              className={classes.tableColumn}
              align={alignment}
              key={columnDetails.attribute}
            >
              <TableSortLabel
                active={orderBy === columnDetails.attribute}
                direction={order}
                onClick={() =>
                  handleSorting(
                    columnDetails.attribute,
                    columnDetails.attributeType
                  )
                }
              >
                {columnDetails.label}
              </TableSortLabel>
            </TableCell>
          );
        })}
      </TableRow>
    </TableHead>
  );
}

export default function ScoreTable({ peopleData }) {
  const classes = useStyles();
  const [shownRows, setShownRows] = React.useState(10);
  const [order, setOrder] = React.useState(undefined);
  const [orderBy, setOrderBy] = React.useState(undefined);
  const [sortedTable, setSortedTable] = React.useState(peopleData);
  React.useEffect(() => {
    setSortedTable(peopleData.slice(0, shownRows));
  }, [peopleData, shownRows]);

  const handleSeeMore = () => {
    setShownRows(shownRows + 10);
  };
  const handleSorting = (attribute, attributeType) => {
    const newOrder =
      attribute !== orderBy ? order : order === "asc" ? "desc" : "asc";
    setOrder(newOrder);
    setOrderBy(attribute);
    setSortedTable(
      getSortedTable(peopleData, newOrder, attribute, attributeType)
    );
  };
  return (
    <React.Fragment>
      <Title>Scores listing</Title>
      <Table size="small">
        <EnhancedHeaderTable
          columnsDetails={columnsDetails}
          handleSorting={handleSorting}
          order={order}
          orderBy={orderBy}
        />
        <TableBody>
          {sortedTable.slice(0, shownRows).map(row => (
            <TableRow key={row.id}>
              <TableCell>
                {row.last_name}
              </TableCell>
              <TableCell>
                {row.first_name}
              </TableCell>
              <TableCell>
                {row.gender}
              </TableCell>
              <TableCell>{row.city}</TableCell>
              <TableCell>
                {row.country}
              </TableCell>
              <TableCell align="right">{row.score}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
      <div className={classes.seeMore}>
        <Link color="primary" href="#" onClick={handleSeeMore}>
          See more scores
        </Link>
      </div>
    </React.Fragment>
  );
}
