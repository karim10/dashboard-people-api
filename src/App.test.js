import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from './App';
import ScoreTable from './ScoreTable';
import { getSortedTable, getChartDataByCountry, getChartDataByGender } from './helpers';
import {
    mockPopleDataWithLength3,
    mockPeopleDataSortedAscByFirstNameWithLength3,
    mockPopleDataSortedDescByScoreWithLength3,
    mockPeopleDataWithLength10,
    mockChartDataGroupedByGender,
    mockAverageScoreByCountry
} from './mocks';
import Chart from './Chart';
import { TableSortLabel } from '@material-ui/core';

// configuration
configure({ adapter: new Adapter() });
function createNodeMock() {
    const doc = document.implementation.createHTMLDocument();
    return { parentElement: doc.body };
}

describe('tests rendering', () => {
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation(init => [init, setState]);
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders ScoreTable', () => {
        const component = renderer.create(<ScoreTable peopleData={mockPeopleDataWithLength10} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders Chart grouped by gender', () => {
        const component = renderer.create(
            <Chart chartType={'gender'} chartData={mockChartDataGroupedByGender} />,
            { createNodeMock }
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
    it('calls handleSorting when header cell is clicked', () => {
        const component = mount(<ScoreTable peopleData={mockPeopleDataWithLength10} />);
        component
            .find(TableSortLabel)
            .findWhere(tsl => tsl.props().sortingkey === 'last_name')
            .first()
            .simulate('click');
        expect(setState).toHaveBeenCalled();
    });
});

describe('tests helpers functions', () => {
    it('sorts people data', () => {
        expect(
            getSortedTable(mockPopleDataWithLength3, {
                orderDirection: 'asc',
                orderBy: 'first_name',
                isNumeric: false
            })
        ).toEqual(mockPeopleDataSortedAscByFirstNameWithLength3);
        expect(
            getSortedTable(mockPopleDataWithLength3, {
                orderDirection: 'desc',
                orderBy: 'score',
                isNumeric: true
            })
        ).toEqual(mockPopleDataSortedDescByScoreWithLength3);
    });

    // it('caculates average score by country', () => {
    //     expect(getChartDataByCountry(mockPeopleDataWithLength10)).toEqual(
    //         mockAverageScoreByCountry
    //     );
    // });
    it('caculates average score by gender', () => {
        expect(getChartDataByGender(mockPeopleDataWithLength10)).toEqual(
            mockChartDataGroupedByGender
        );
    });
});
