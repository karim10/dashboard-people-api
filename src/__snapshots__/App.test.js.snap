// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests rendering renders Chart grouped by gender 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  >
    Score statistics
  </h2>,
  <div
    className="recharts-responsive-container"
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "display": "none",
          "height": 0,
          "position": "absolute",
          "visibility": "hidden",
          "width": 0,
        }
      }
    />
  </div>,
]
`;

exports[`tests rendering renders ScoreTable 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  >
    Scores listing
  </h2>,
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-searchFiled-352"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink={[Function]}
    >
      Search...
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
      <div
        className="MuiInputAdornment-root"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiPaper-elevation0 makeStyles-paper-351 MuiPaper-rounded"
  >
    <div
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    >
      <div
        aria-colcount={6}
        aria-rowcount={10}
        className="ReactVirtualized__Table"
        role="grid"
        style={Object {}}
      >
        <div
          className="ReactVirtualized__Table__headerRow makeStyles-flexContainer-497"
          role="row"
          style={
            Object {
              "height": 48,
              "overflow": "hidden",
              "paddingRight": 0,
              "width": 0,
            }
          }
        >
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 250px",
                "flex": "0 1 250px",
                "msFlex": "0 1 250px",
              }
            }
          >
            <div
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head makeStyles-tableCell-498 makeStyles-flexContainer-497 MuiTableCell-alignLeft"
              style={
                Object {
                  "height": 48,
                }
              }
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                sortingkey="last_name"
                tabIndex={0}
              >
                Last Name
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 250px",
                "flex": "0 1 250px",
                "msFlex": "0 1 250px",
              }
            }
          >
            <div
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head makeStyles-tableCell-498 makeStyles-flexContainer-497 MuiTableCell-alignLeft"
              style={
                Object {
                  "height": 48,
                }
              }
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                sortingkey="first_name"
                tabIndex={0}
              >
                First Name
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 250px",
                "flex": "0 1 250px",
                "msFlex": "0 1 250px",
              }
            }
          >
            <div
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head makeStyles-tableCell-498 makeStyles-flexContainer-497 MuiTableCell-alignLeft"
              style={
                Object {
                  "height": 48,
                }
              }
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                sortingkey="gender"
                tabIndex={0}
              >
                Gender
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 250px",
                "flex": "0 1 250px",
                "msFlex": "0 1 250px",
              }
            }
          >
            <div
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head makeStyles-tableCell-498 makeStyles-flexContainer-497 MuiTableCell-alignLeft"
              style={
                Object {
                  "height": 48,
                }
              }
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                sortingkey="city"
                tabIndex={0}
              >
                City
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 250px",
                "flex": "0 1 250px",
                "msFlex": "0 1 250px",
              }
            }
          >
            <div
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head makeStyles-tableCell-498 makeStyles-flexContainer-497 MuiTableCell-alignLeft"
              style={
                Object {
                  "height": 48,
                }
              }
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                sortingkey="country"
                tabIndex={0}
              >
                Country
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            className="ReactVirtualized__Table__headerColumn"
            role="columnheader"
            style={
              Object {
                "WebkitFlex": "0 1 250px",
                "flex": "0 1 250px",
                "msFlex": "0 1 250px",
              }
            }
          >
            <div
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head makeStyles-tableCell-498 makeStyles-flexContainer-497 MuiTableCell-alignRight"
              style={
                Object {
                  "height": 48,
                }
              }
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                sortingkey="score"
                tabIndex={0}
              >
                Score
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          aria-label="grid"
          aria-readonly={true}
          className="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
          onScroll={[Function]}
          role="rowgroup"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "inherit",
              "height": -48,
              "overflowX": "hidden",
              "overflowY": "auto",
              "position": "relative",
              "width": 0,
              "willChange": "transform",
            }
          }
          tabIndex={0}
        />
      </div>
    </div>
  </div>,
]
`;
